schema.ts

```diff
@@ -1232,6 +1232,82 @@ export interface ObjectSchemaDefinition<K extends string, L extends string>
   // TODO(dweitzman): Only support options in the typing when the codaType is ValueHintType.SelectList.
 }
 
+/**
+ * Discriminent used to determine the permission type
```

Comment:
Would probably just simplify to something like "The type of principal that can be applied to a permission."